cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt

```diff
@@ -511,36 +504,83 @@ private class CampaignManagerImpl(
+
+        return ((isGdprPresent || isCcpaUuidPresent || isUsNatUuidPresent) && isLocalStateEmpty) ||
+            isV630LocalStatePresent ||
+            isV690LocalStatePresent ||
+            storedCcpaWithoutGPP ||
+            usnatApplicableSectionChanged ||
+            ccpa2usnat ||
+            authId != null
+    }
+
+    override fun consentStatusLog(authId: String?) {
+        if (logger == null) return
+        val isGdprPresent = dataStorage.gdprConsentUuid != null
+        val isCcpaUuidPresent = dataStorage.ccpaConsentUuid != null
+        val isUsNatUuidPresent = usNatConsentData?.uuid != null
+        val isLocalStateEmpty = messagesOptimizedLocalState?.jsonObject?.isEmpty() == true
+        val isV630LocalStatePresent = dataStorage.preference.all.containsKey(LOCAL_STATE)
+        val isV690LocalStatePresent = dataStorage.preference.all.containsKey(LOCAL_STATE_OLD)
+        val ccpa2usnat = (
+            ccpaConsentStatus != null &&
+                usNatConsentData == null &&
+                spConfig.hasTransitionCCPAAuth() &&
+                spConfig.isIncluded(USNAT)
+            )
+        val storedCcpaWithoutGPP: Boolean = ccpaConsentStatus
+            ?.let { it.gppData == null || it.gppData.isEmpty() } ?: false
+
+        val shouldCallConsentStatus =
+            ((isGdprPresent || isCcpaUuidPresent || isUsNatUuidPresent) && isLocalStateEmpty) ||
+                isV630LocalStatePresent ||
+                isV690LocalStatePresent ||
+                usnatApplicableSectionChanged ||
+                ccpa2usnat ||
+                authId != null ||
+                storedCcpaWithoutGPP
+
+        logger.computation(
+            tag = "shouldCallConsentStatus",
+            msg = """ shouldCallConsentStatus[$shouldCallConsentStatus]
+            """.trimIndent(),
+            json = JSONObject().apply {
+                put(
+                    "consentsStoredDetails",
+                    JSONObject().also {
+                        it.put("(GdprUuid ", isGdprPresent)
+                        it.put("OR  CcpaUuid", isCcpaUuidPresent)
+                        it.put("OR  UsnatUuid)", isUsNatUuidPresent)
+                        it.put("AND isLocalStateEmpty", isLocalStateEmpty)
+                    }
+                )
+                put(" consentsStored", ((isGdprPresent || isCcpaUuidPresent || isUsNatUuidPresent) && isLocalStateEmpty))
+                put(" OR isV630LocalStatePresent", isV630LocalStatePresent)
+                put(" OR isV690LocalStatePresent", isV690LocalStatePresent)
+                put(" OR usnatApplicableSectionChanged", usnatApplicableSectionChanged)
+                put(" OR storedCcpaWithoutGPP", storedCcpaWithoutGPP)
+                put(" OR transitionCcpa2Usnat", ccpa2usnat)
+                put(" OR authId", authId != null)
+                put("return shouldCallConsentStatus", shouldCallConsentStatus)
+            }
+        )
+    }
```

Comment:
@carmelo-iriti could you consider removing the method above? That's adding 50 lines of code for something that does not contribute for the function of the SDK. If not removing, at least "hiding" the logging logic somewhere else that doesn't interfere with the readability of the SDK classes.